pkg/operator/ceph/config/conditions.go

```diff
@@ -172,21 +175,25 @@ func conditionMapping(conditions []cephv1.Condition) {
 }
 
 // CheckConditionReady checks whether the cluster is Ready and returns the message for the Progressing ConditionType
-func CheckConditionReady(context *clusterd.Context, namespace, name string) string {
-	cluster, err := context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
-	if err != nil {
-		logger.Errorf("failed to get cluster %v", err)
-	}
-	if cluster.Status.Conditions != nil && len(conditionMap) == 0 {
-		conditionMapping(cluster.Status.Conditions)
+func CheckConditionReady(context *clusterd.Context, namespace, name, service string) string {
+	if name != "" {
```

Comment:
Won't `name` always be set? If not, I'd suggest a comment to indicate why sometimes it is set or not.

Reply:
Hmm its already set everywhere. But when running the unit tests for osd package, everything fails stating the name is blank and I am getting hit with a nil pointer error. I put this condition especially to avoid the testing.

Reply:
Looks like on line 182 you need to return after the error, otherwise you would crash. If that is fixed, you shouldn't need the check for `name != ""`.